src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2477,6 +2479,15 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             CancellationToken cancellationToken)
         {
             options = options ?? EmitOptions.Default.WithIncludePrivateMembers(metadataPEStream == null);
+
+            if (options.DefaultSourceFileEncoding is null)
+            {
+                // If there's no source encoding default already provided, pull it out from the 
+                // first source that was compiled. This encoding depends on the fallback logic for parsing
+                // text and needs to be stored if available. SourceText parsed
+                options = options.WithDefaultSourceFileEncoding(SyntaxTrees.FirstOrDefault()?.Encoding);
+            }
```

Comment:
Not sure why this is needed.

Reply:
I rolled through this a few times. `fallbackSourceFileEncoding` is something that's informed by a compilation. Specifically, if we store on `EmitOptions` that means that the act of doing a compilation changes the `EmitOptions` that were passed into the compilation, which seemed to be wrong. Is there a way `fallbackSourceFileEncoding` makes sense to pass in for something that wouldn't be determined by a compilation?

Reply:
I'm not quite sure what you mean. I think if once you follow up on the other comment this won't be needed.

Reply:
You're right, I see now. Value is retrieved from common compilation directly now, this is not needed.

Reply:
Ok, value is now added to `CommonCompilation`, and passed through `EmitOptions` directly. I added `EmitOptions` to the modulebuilder as well to avoid some problems with having to update all of the duplicated fields.